packages/shared/src/lib/core/wallet/actions/send/createEvmChainToEvmChainTransaction.ts

```diff
@@ -58,12 +58,18 @@ export function createEvmChainToEvmChainTransaction(
         if (!nft) {
             throw new Error(localize('error.send.invalidSendParameters'))
         }
-        destinationAddress = getDestinationAddressForNft(nft, recipientAddress)
+        destinationAddress = getDestinationAddressForNft(nft)
     }
 
     let data: string | undefined
     if (token?.standard === TokenStandard.Irc30 || token?.standard === TokenStandard.Erc20 || nft) {
-        data = getDataForTransaction(chain, originAddress, recipientAddress, token, amount, nft)
+        data =
+            token && amount
+                ? getTokenDataForTransaction(chain, recipientAddress, token, amount)
+                : nft
+                ? getNftDataForTransaction(chain, originAddress, recipientAddress, nft)
+                : undefined
+
```

Comment:
Clean up this part by moving it up to the if else branch above and extract a function that sets the amount to '0' if smart contract call and otherwise throws an error.